roles/openshift_facts/library/openshift_facts.py

```diff
@@ -1917,8 +1917,8 @@ def __init__(self, role, filename, local_facts,
             self.system_facts = ansible_facts(module, ['hardware', 'network', 'virtual', 'facter'])  # noqa: F405
             for (k, v) in self.system_facts.items():
                 self.system_facts["ansible_%s" % k.replace('-', '_')] = v
-        except UnboundLocalError:
-            # ansible-2.2
+        except (UnboundLocalError, RuntimeError):
```

Comment:
Ah, I see what's happening now. If you can, I'd do something like this:

```python
additional_facts = {}
for (k, v) in self.system_facts.items():
    additional_facts["ansible_%s" % k.replace('-', '_')] = v

self.system_facts.update(additional_facts)
```

That will get those new items added to the ``system_facts`` without modifying the structure being iterated over.